### PR TITLE
Refactor CI workflows

### PR DESCRIPTION
**Pull request by christopherwharrop-noaa**
This update refactors the CI workflows to improve their efficiency and also to add critical tests for Parsl and Flux with MPI in a Slurm cluster environment.  Previous CI workflows used a complex multi-stage container strategy to build the entire ExaWorks tool set (flux, radical, parsl, and stc) using Spack.  This tested install scripts for those tools, but took many hours to run and didn't allow for testing of Parsl and Flux applications in a cluster environment.  Because the previous CI tests are intractable and don't allow for the type of functional testing we need, they are being discarded. Additionally, we have also decided to focus efforts on integration of Parsl and Flux and have put explorations of Radical and Swift/T on the back burner for now.

We recently discovered that Parsl must be install with `pip` instead of Spack because that is the only way to obtain the later versions which are required when using it with Flux.  We've also realized that we need to test with the Intel compilers.  And so the installation strategy has changed.  The new CI builds a containerized Slurm cluster that has the Intel OneAPI compilers, Spack, Flux, and Parsl installed into it.  The CI then uses that container to run Parsl and Flux tests in a Slurm cluster environment with MPI. The test scripts are mounted via a volume into the container so that the container doesn't need to be updated as more tests are added.

### CODE REVIEW

File: docker/master/docker-entrypoint.sh
Comment (NaureenBharwaniNOAA) [inline docker/master/docker-entrypoint.sh:1]: This file seems to have overlapping lines of code that are included in the docker/node/docker-entrypoint.sh file. The only difference I see is on line 7 in both corresponding files. Is there a way to consolidate these files?

**Review by NaureenBharwaniNOAA — COMMENTED** [review]
Besides the overlapping content in the two docker-entrypoint.sh files, everything looks pretty concise to me.

File: docker/master/docker-entrypoint.sh
Comment (christopherwharrop-noaa) [reply docker/master/docker-entrypoint.sh:1]: Good eye.  Turns out the docker entry points were leftover when copying from the DockerSlurm repository as a starting point.  They have all been removed.  Thank you for making note of that.  Exactly what I needed.

**Review by NaureenBharwaniNOAA — APPROVED** [review]
Glad I could help. Looks good to me!

**Comment by christopherwharrop-noaa** [comment]
@NaureenBharwaniNOAA - Thank you for the approval.  Once the CI test passes, I will merge it. If more changes are necessary to make it work, I'll need to request another review.

**Comment by christopherwharrop-noaa** [comment]
@NaureenBharwaniNOAA - I **finally** figure out how to get this all to work correctly.  The docker layer caching with Github Actions should all be working now.  And the full MPI Parsl/Flux tests are passing too.  Please take another look when you have a moment.

File: .github/workflows/docker-slurm.yml
Comment (NaureenBharwaniNOAA) [inline .github/workflows/docker-slurm.yml:93]: Confirming that leaving this in is the right option. Could we potentially move this code block to its only file? How will the user interactive with this code block? Will this require the user to uncomment this section or is this just for internal use?

File: docker/test/parsl_mpi_flux_hello.py
Comment (NaureenBharwaniNOAA) [inline docker/test/parsl_mpi_flux_hello.py:49]: I only see one reference to the resource_list function here and am unsure exactly what the use of this function is? It doesn't look like anything is being done within the function or returned from it. Do we still need to implement this? Unsure of the functionality here. Are we returning this docstring?

**Review by NaureenBharwaniNOAA — COMMENTED** [review]
Overall the code looks good to me. The comments I added should be easily resolved with a bit more understanding of what actions the code blocks are performing.

File: .github/workflows/docker-slurm.yml
Comment (christopherwharrop-noaa) [reply .github/workflows/docker-slurm.yml:93]: This step is only useful when the github actions is failing and the reason cannot be determined without gaining interactive access to the runner for debugging purposes.  The reason I leave it in here commented out is because it's a recipe that isn't at all obvious and which will likely be needed in the future due to complexities of Slurm/Parsl/Flux.  I could remove it, but then, I'd have to remember it from scratch the next time it is needed or store it somewhere else.  Although it's not usually a good idea to leave commented code, this is a GHA workflow step that is likely to be needed repeatedly in the future and it's much simpler to just leave it here.

File: docker/test/parsl_mpi_flux_hello.py
Comment (christopherwharrop-noaa) [reply docker/test/parsl_mpi_flux_hello.py:49]: The `flux resource list` command is a Flux command that asks Flux to print out the resources it thinks it has access to.  This is a simple test in the Parsl workflow that verifies that the resources we expect to be available are also seen by Flux.  The use of a returned docstring is the idiomatic way to implement Parsl workflow "Apps".  The contents of the `resource_list.txt` file contains the output of the command and the CI probably should actually check that the file is as expected.

**Review by NaureenBharwaniNOAA — APPROVED** [review]
Thanks for the clarification, those code blocks make sense to me then. LGTM!

**Pull request merged**
